RecoLocalCalo/HGCalRecAlgos/plugins/alpaka/HGCalRecHitCalibrationESProducer.cc

```diff
@@ -68,6 +68,18 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
         std::copy(begin, end, &column_SoA[offset]);
       }
 
+      template <typename T, typename P>
+      static void fill_SoA_eigen_row(P& soa, const std::vector<std::vector<T>>& values, const size_t row) {
+        if (row >= values.size())
+          edm::LogWarning("HGCalCalibrationESProducer")
+              << " Tried to copy row " << row << " to SoA, but only have " << values.size() << " values in JSON!";
+        if (!values.empty() && int(values[0].size()) != soa.size())
+          edm::LogWarning("HGCalCalibrationESProducer")
+              << " Expected " << soa.size() << " elements in Eigen vector, but got " << values[0].size() << "!";
```

Comment:
If these checks fire, the result should be something else than executing the loop below with undefined behavior.

Reply:
I changed to throw edm::Exception